projects/libsrtp/project.yaml

```diff
@@ -0,0 +1,6 @@
+homepage: "https://github.com/cisco/libsrtp"
+primary_contact: "richbarn@cisco.com"
+auto_ccs:
+    - "guidovranken@gmail.com"
+fuzzing_engines:
+    - libfuzzer
```

Comment:
Does this break with AFL? If not let's remove this.

Reply:
ping @guidovranken was there a reason not to use AFL here?
I understand if you don't want to spend more time on this integration.

Reply:
I'm sorry. I copied this over from bignum-fuzzer which has the same lines. I can't remember why I put that there in the first place, but I see no reason why AFL or honggfuzz wouldn't work, as long as they call ```LLVMFuzzerTestOneInput```, which the oss-fuzz platform takes care of I assume.

So I would like to remove the libFuzzer restriction from both libsrtp and bignum-fuzzer. Is there a way to force ```python infra/helper.py build_fuzzers``` to build against AFL/honggfuzz? So I can check if there are any unexpected problems.

Should I make 1 or 2 separate PR's for fixing libsrtp and bignum-fuzzer?

Reply:
>I'm sorry. I copied this over from bignum-fuzzer which has the same lines.

No problem I figured that is what happened.

>I can't remember why I put that there in the first place,

It sort of makes sense to me to have there. I don't exactly know those targets, but I assume the inputs are basically just numbers so different mutation techniques and other engine differences probably don't help much. Sort of similar to why only ASAN is used. Just guessing though =). We don't fully understand what benefits AFL brings in addition to LF but it definitely seems to help in general, and particularly for targets that can be slow or OOM (since these, [for the time being](https://github.com/google/oss-fuzz/issues/2178), stop libFuzzer in its tracks).
Feel free to remove it from bignum-fuzzer too. 

>but I see no reason why AFL or honggfuzz

FYI we don't actually use honggfuzz on ClusterFuzz, our documentation is basically incorrect to compare honggfuzz and AFL (see [this part](https://github.com/google/oss-fuzz/issues/611#issuecomment-463795298) of the issue tracking it) I'm going to change that right now. We do use AFL on ClusterFuzz though so if AFL builds fail then the build will fail.

>as long as they call `LLVMFuzzerTestOneInput`, which the oss-fuzz platform takes care of I assume.

Right, we use [a harness for AFL](https://github.com/llvm/llvm-project/blob/master/compiler-rt/lib/fuzzer/afl/afl_driver.cpp) that calls `LLVMFuzzerTestOneInput` and communicates with `afl-fuzz`.

>Is there a way to force `python infra/helper.py build_fuzzers` to build against AFL/honggfuzz? So I can check if there are any unexpected problems.

You just need to add `--engine afl` to the command to build with AFL (btw we only use AFL with ASAN, no need to test other sanitizers).

In general though, AFL fuzzing should "just work", I assume most people dont do any testing on AFL builds.

>Should I make 1 or 2 separate PR's for fixing libsrtp and bignum-fuzzer?

One PR is fine.

Reply:
Both projects build fine with AFL but I get this when I try to run a fuzzer (any fuzzer in either project):

```
$ python infra/helper.py run_fuzzer libsrtp srtp-fuzzer
Running: docker run --rm -i --privileged -e FUZZING_ENGINE=libfuzzer -e SANITIZER=address -e RUN_FUZZER_MODE=interactive -v /home/jhg/ossfuzz-libmpdec/oss-fuzz/build/out/libsrtp:/out -t gcr.io/oss-fuzz-base/base-runner run_fuzzer srtp-fuzzer
Using seed corpus: srtp-fuzzer_seed_corpus.zip
/out/srtp-fuzzer -rss_limit_mb=2048 -timeout=25 /tmp/srtp-fuzzer_corpus < /dev/null
======================= INFO =========================
This binary is built for AFL-fuzz.
To run the target function on individual input(s) execute this:
  /out/srtp-fuzzer < INPUT_FILE
or
  /out/srtp-fuzzer INPUT_FILE1 [INPUT_FILE2 ... ]
To fuzz with afl-fuzz execute this:
  afl-fuzz [afl-flags] /out/srtp-fuzzer [-N]
afl-fuzz will run N iterations before re-spawning the process (default: 1000)
======================================================
Reading 18446744073709551615 bytes from -rss_limit_mb=2048
terminating with uncaught exception of type std::length_error: vector
AddressSanitizer:DEADLYSIGNAL
=================================================================
==14==ERROR: AddressSanitizer: ABRT on unknown address 0x00000000000e (pc 0x7f816058c428 bp 0x7ffc6edbb5e0 sp 0x7ffc6edbb3a8 T0)
SCARINESS: 10 (signal)
    #0 0x7f816058c427 in gsignal (/lib/x86_64-linux-gnu/libc.so.6+0x35427)
    #1 0x7f816058e029 in abort (/lib/x86_64-linux-gnu/libc.so.6+0x37029)
    #2 0x5b3e96 in abort_message (/out/srtp-fuzzer+0x5b3e96)
    #3 0x5b7ef2 in demangling_terminate_handler() (/out/srtp-fuzzer+0x5b7ef2)
    #4 0x5b3805 in std::__terminate(void (*)()) (/out/srtp-fuzzer+0x5b3805)
    #5 0x5b59dc in __cxa_throw (/out/srtp-fuzzer+0x5b59dc)
    #6 0x5a6e95 in std::__1::__vector_base_common<true>::__throw_length_error() const (/out/srtp-fuzzer+0x5a6e95)
    #7 0x539504 in __vallocate /usr/local/bin/../include/c++/v1/vector:975:15
    #8 0x539504 in vector /usr/local/bin/../include/c++/v1/vector:1119
    #9 0x539504 in ExecuteFilesOnyByOne(int, char**) /src/libfuzzer/afl/afl_driver.cpp:153
    #10 0x539671 in main /src/libfuzzer/afl/afl_driver.cpp:193:12
    #11 0x7f816057782f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)
    #12 0x41cf18 in _start (/out/srtp-fuzzer+0x41cf18)
```

Is this expected when running AFL locally or something?

Reply:
Heh, I guess this is unexpected but it makes sense why you get that error, the helper script is passing libFuzzer arguments to the target.

This won't happen in "production" though, so feel free to land the change.
My guess is no one has ever used this command for AFL (or at least no one who would file a bug).

Reply:
Filed #2199 to fix this.

Reply:
Actually, there's no bug here, I think you may need to pass `--engine afl` to `run_fuzzer` as well.